packages/embark/src/lib/core/config.js

```diff
@@ -280,16 +280,17 @@ Config.prototype._getFileOrObject = function(object, filePath, property) {
   return dappPath(object, filePath);
 };
 
-/*eslint complexity: ["error", 30]*/
+/*eslint complexity: ["error", 34]*/
 Config.prototype.loadBlockchainConfigFile = function() {
   const blockchainDefaults = getBlockchainDefaults(this.env);
   const configFilePath = this._getFileOrObject(this.configDir, 'blockchain', 'blockchain');
 
   const userConfig = this._loadConfigFile(configFilePath, blockchainDefaults, true);
   const envConfig = userConfig[this.env];
-
+  
   if (envConfig) {
-    if (envConfig.ethereumClientName || envConfig.hasOwnProperty('isDev') || envConfig.hasOwnProperty('mineWhenNeeded')) {
+    const isCustomBlockchain = ![constants.blockchain.clients.geth, constants.blockchain.clients.parity].includes(envConfig.client);
+    if (envConfig.ethereumClientName || (!isCustomBlockchain && envConfig.hasOwnProperty('isDev')) || envConfig.hasOwnProperty('mineWhenNeeded')) {
```

Comment:
Don't think it makes sense to change this.
This condition was to check if people used the old `isDev` property in the config, which has now been changed to `miningMode: 'dev'`

Reply:
Right, so for nodes without mining, ie POS (like lightchain), it doesn't make sense to include a `miningMode: 'dev'` in the blockchain config to denote a development environment. (In fact, it's not very clear in the first place). So, for custom blockchain plugins, we need a way to denote that a particular environment is a development environment, and to do this, we are allowing them to use `isDev: true` in the blockchain config. For internal clients (node/geth), we are still going to show the error that you put in regarding `isDev` being removed.

This could be removed in favour of letting the blockchain plugin handle `this.isDev = blockchainConfig.client === "development"` if it's too confusing.

Reply:
Ah ok, I understand your point, not sure what's the best approach here, but maybe we can keep `isDev` for plugins for now